### PR TITLE
Feature/associations

### PR DESCRIPTION
**Pull request by HFG43**
Hi there! I'm sending this pull request for you to review our code. Thanks in advance for your feedback!

Milestones 🎯:
📌 Implement book, classroom, and rental new classes
📌 Implement associations
📌 Test the app in the command line

🎖️ General requirements:
Gitflow
No linter errors
Ensure the use of best practices and Ruby style

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+  end
```
Comment (VICTOR-CHUKWU) [inline book.rb:13]: - [ ] this method should take person and date as parameters and create a new instance of rental achieving the 'has-many' association
```
 def add_rental(person, date)
    Rental.new(date, self, person)
  end
```

File: person.rb [hunk 2]
```diff
@@ -29,4 +34,8 @@ def of_age?
   def can_use_servicies?
     of_age? || @parent_permission
   end
+
+  def add_rental(rental)
+    @rentals.push(rental) unless @rentals.include?(rental)
+  end
```
Comment (VICTOR-CHUKWU) [inline person.rb:40]: - [ ] This method should take book and date as parameters and create a new instance of rental achieving the 'has-many' association
```
 def add_rental(book, date)
    Rental.new(date, self, book)
  end
```

**Review by VICTOR-CHUKWU — CHANGES_REQUESTED** [review]
Hi @HFG43  👋  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

almost

## Highlights
- Good PR
- Good commit
- association created between class


## Required Changes ♻️

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 2 above)
Comment (HFG43) [reply person.rb]: Thanks @VICTOR-CHUKWU! I have applied the required changes

**Review by DuaneDave — APPROVED** [review]
Hi @HFG43 👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✔️ Associations are correctly set up.Good job 👍 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**